mods/Seasonal/moves.js

```diff
@@ -0,0 +1,97 @@
+"use strict";
+
+exports.BattleMovedex = {
+
+"growth": {
+		accuracy: true,
+		basePower: 0,
+		category: "Status",
+		desc: "Raises the user's Attack and Special Attack by 1 stage. If the weather is Sunny Day, raises the user's Attack and Special Attack by 2 stages.",
+		shortDesc: "Raises user's Attack and Sp. Atk by 1; 2 in Sun.",
+		id: "growth",
+		name: "Growth",
+		inherit: true,
+		pp: 20,
+		priority: 0,
+		flags: {snatch: 1},
+		onModifyMove: function (move) {
+		switch (this.random(7)) {
+				case 0:
+					move.category = 'Special';
+					move.type = 'Fire';
+					move.basePower = 200;
+					move.onTryHit = function () {
+						this.add('-message', "Pumpkin bomb!");
+					};
+					move.onHit = function () {};
+					break;
+				case 1:
+					move.category = 'Physical';
+					move.type = 'Poison';
+					move.basePower = 25;
+					move.multihit = 4;
+					move.onTryHit = function () {
+						this.add('-message', "Toilet paper missile attack!");
+					};
+					move.onHit = function () {};
+					break;
+				case 2:
+					move.onTryHit = function () {
+						this.add('-message', "Yum! Chocolate!");
+					};
+					move.onHit = function (target, source) {
+						this.heal(Math.ceil(target.maxhp * 0.5));
+					};
+					break;
+				case 3:
+					move.onTryHit = function () {
+						this.add('-message', "This is a rather bland candy.");
+					};
+					move.onHit = function (target, source) {
+						this.heal(Math.ceil(target.maxhp * 0.25));
+						target.setStatus('par');
+						target.addVolatile('confusion');
+					};
+					break;
+				case 4:
+					move.onTryHit = function () {
+						this.add('-message', "You are about to be rotten-egged on!");
+					};
+					move.onHit = function (target, source) {
+						target.setStatus('tox');
+						target.addVolatile('torment');
+					};
+					break;
+				case 5:
+					move.category = 'Special';
+					move.type = 'Dark';
+					move.basePower = 500;
+					move.self = {volatileStatus: 'mustrecharge'};
+					move.onTryHit = function () {
+						this.add('-message', "Ultimate Super Hiper Mega Awesome Beam destroyer of worlds!");
+					};
+					move.onHit = function (target, source) {
+						this.add('-message', source.name + " was caught in the explosion!");
+						source.setStatus('brn');
+						source.addVolatile('disabled');
+						source.addVolatile('confusion');
+					};
+					break;
+				case 6:
+					move.onTryHit = function () {
+						this.add('-message', "Have some refreshment, my fellow.");
+					};
+					move.onHit = function (target, source) {
+						target.addVolatile('aquaring');
+					};
+					break;
+				      }
+		}
+},
+		secondary: false,
```

Comment:
Tbh since you set inherit to true all these lines are unnecessary, once you set an inherit true only change what's different from the original move (i.e. a PP change, BP change, etc)